fontforgeexe/cvpalettes.c

```diff
@@ -1558,6 +1559,50 @@ static BDFChar *BDFCharFromLayer(SplineChar *sc,int layer) {
 return( SplineCharAntiAlias(&dummy,ly_fore,24,4));
 }
 
+/**
+ *  \brief Recalculate the number of visible layers and reposition the scrollbar
+ *
+ *  \param [in] cv The charview
+ */
+static void CVLayers2Reflow(CharView *cv) {
+    extern int _GScrollBar_Width;
+    GGadget *scrollbar;
+    GRect cvl2size;
+
+    GDrawGetSize(cvlayers2, &cvl2size);
+    // Magic! I verified symmetry in Paint.NET
+    layer2.header_height = GGadgetGetY(GWidgetGetControl(cvlayers2, CID_VGrid)) - 1;
+    // Minus 2 because we always have the 'Guide' and 'Back' entries
+    layer2.visible_layers = (cvl2size.height - layer2.header_height - 2 * CV_LAYERS2_LINE_HEIGHT) / CV_LAYERS2_LINE_HEIGHT;
+    if (layer2.visible_layers < 0) {
+        layer2.visible_layers = 0;
+    }
+
+    scrollbar = GWidgetGetControl(cvlayers2, CID_SB);
+    // Check if we need the scrollbar or not.
+    if (layer2.current_layers - 2 <= layer2.visible_layers) {
+        GGadgetSetVisible(scrollbar, false);
+        layer2.sb_start = cvl2size.width;
+        layer2.offtop = 0;
+    } else {
+        layer2.sb_start = cvl2size.width - GDrawPointsToPixels(cv->gw, _GScrollBar_Width);
+        GGadgetMove(scrollbar, layer2.sb_start, 0);
+        GGadgetResize(scrollbar, GDrawPointsToPixels(cv->gw, _GScrollBar_Width), cvl2size.height);
+        GGadgetSetVisible(scrollbar, true);
+
+        GScrollBarSetBounds(scrollbar, 0, layer2.current_layers - 2, layer2.visible_layers);
+        if (layer2.offtop > layer2.current_layers - layer2.visible_layers) {
+            layer2.offtop = layer2.current_layers - layer2.visible_layers;
+        }
+        if (layer2.offtop < 0) {
+            layer2.offtop = 0;
+        }
```

Comment:
Could you explain the math here a bit more? I thought that current_layers and visible_layers were not compatible numbers since current_layers includes the non-scrolling layers, and visible_layers excludes them.